Improve the page on the incremental import (#1294)
neo4j/docs-operations: modules/ROOT/pages/tools/neo4j-admin/neo4j-admin-import.adoc

Incremental import is not supported for `block` format yet.
====

Incremental import allows you to incorporate large amounts of data in batches into the graph.
You can run this operation as part of the initial data load when it cannot be completed in a single full import.
Besides, you can update your graph by importing data incrementally, which is more performant than transactional insertion of such data.

Incremental import requires the use of `--force` and can be run on an existing database only.

You must stop your database, if you want to perform the incremental import within one command.

If you cannot afford a full downtime of your database, split the operation into several stages:

> * _prepare_ stage

Reviewer: ```suggestion
* _prepare_ stage (offline)
```

> * _build_ stage

Reviewer: ```suggestion
* _build_ stage (offline or read-only)
```

> * _merge_ stage

Reviewer: ```suggestion
* _merge_ stagen(offline)
```

> * _merge_ stagen(offline)

Author: ```suggestion
* _merge_ stagen (offline)
```

The database must be stopped for the `prepare` and `merge` stages.

> During the `build` stage the database must not be written to.

Reviewer: ```suggestion
During the `build` stage, the database can be left online but put into read-only mode.
```

> If it is started, you have to put it into read-only mode.

Reviewer: ```suggestion
For a detailed example, see <<incremental-import-stages>>.
```

> For more details, read <<import-tool-incremental-examples, Examples>>.

Reviewer: ```suggestion
```

> Note that your database should have correct uniqueness constraints for the affected nodes, but the tool will tell you and exit, if such is missing.

Reviewer: ```suggestion
[WARNING]
====
Your database must have correct uniqueness constraints for the affected nodes.
Otherwise, the command will throw an error and exit.
====
```

Author: @renetapopova, I decided to proceed with your previous suggestion.😁 Otherwise, we'll get three warnings in one section.

> Note that your database should have correct uniqueness constraints for the affected nodes, but the tool will tell you and exit, if such is missing.

Reviewer: We have a similar warning in the `=== Usage and limitations`  section, which is not tested against. So, I think we could update that by adding something like: "Otherwise, the command throws an error." and remove this one here.

Author: To be honest, I'd prefer to keep this sentence here, in the introductory section. The idea was to explain what we mean by words 'database has to be in precisely the **expected state**'. I'd like to collect all warnings in one place, as users often don't scroll down (unfortunately). 
I agree that there is a similar warning in the section `Usage and limitations` but it's more detailed there with a cross-reference for further information.

> If the _merge_ stage fails, is aborted or crashes it may corrupt the database.

Reviewer: ```suggestion
[WARNING]
====
It is highly recommended to back up your database before running the incremental import, as if the _merge_ stage fails, is aborted, or crashes, it may corrupt the database.
====
```

> 

Reviewer: ```suggestion
```

> Therefore, it is highly recommended to back up your database before running the incremental import.

Reviewer: ```suggestion
```

=== Description

> Incremental import into an existing database.

Reviewer: This description comes from the code base and is tested against it. I think we should add all the explanations at the beginning above or as part of the Examples section.

Author: I didn't know that. So, it means that we cannot change anything in the Description section, doesn't it? Is it worth some sort of explanation made in a format of comments (hidden notes with //)?

> Incremental import requires the use of `--force` and can be run on an existing database only.

Author: Probably it's a good place where we can explain the reason for using this flag.

|===

[[import-tool-incremental-examples]]

> === Two ways to import data incrementally / Two modes of the incremental import

Reviewer: All tool's sections follow the same/similar structure - short introductory sentences, syntax, description, usage, and examples. The idea is that if users know one tool, it will be easier for them to find information for another one. So, I would suggest that we keep the structure as it is. Also, we'll need to change the headings every time we add another example.

Author: Yes, I see that tool's sections are structured the same way and I agree that it makes sense. On the other side, I don't think that the title 'Examples' is informative enough in this particular case. Actually, we're not talking about the examples on how to apply `neo4j-admin` syntax to import data incrementally , but we're talking about two different ways to run the incremental import. I'm suggesting to think on alternatives. If the 'Examples' is the best way, let's keep it.

----
====

> ==== Incremental import in stages

Reviewer: ```suggestion
[[incremental-import-stages]]
==== Incremental import in stages
```

$ bin/neo4j-admin database import incremental --stage=prepare --nodes=N1=../../raw-data/incremental-import/c.csv db1
----
. `build` stage:
+

> During this stage, the import tool imports the data into the database.

Reviewer 2: The `build` stage doesn't quite import the data into the db, rather imports it, deduplicates and validates the data into a temporary location.

Since you mention `new increment path` above, perhaps use that same term here.

> During this stage, the import tool imports the data into the database.

Author: ```suggestion
During this stage, the import tool imports the data, deduplicates it, and validates it in the new increment database path.
```

Author: @tinwelint, is this new sentence more accurate?
